torchrl/objectives/ppo.py

```diff
@@ -466,12 +471,25 @@ def _log_weight(
             self.actor_network
         ) if self.functional else contextlib.nullcontext():
             dist = self.actor_network.get_dist(tensordict)
-        log_prob = dist.log_prob(action)
 
         prev_log_prob = tensordict.get(self.tensor_keys.sample_log_prob)
         if prev_log_prob.requires_grad:
             raise RuntimeError("tensordict prev_log_prob requires grad.")
 
+        if isinstance(dist, CompositeDistribution):
```

Comment:
ditto, isn't there a way to test if the dist returns td log-probs other than checking the id of the dist?